common/persistence/pinotUserQueryTimeFilter.go

```diff
@@ -0,0 +1,172 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package persistence
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/xwb1989/sqlparser"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/definition"
+	"github.com/uber/cadence/common/types"
+)
+
+// VisibilityQueryFilter for sql query validation
+type VisibilityQueryFilter struct {
+}
+
+// NewVisibilityQueryFilter create VisibilityQueryValidator
+func NewVisibilityQueryFilter() *VisibilityQueryFilter {
+	return &VisibilityQueryFilter{}
+}
+
+// FilterQuery validates that search attributes in the query and returns modified query.
+func (qv *VisibilityQueryFilter) FilterQuery(whereClause string) (string, error) {
```

Comment:
Can we reuse queryValidator here, it is the same code

Reply:
not quite. This one is a much easier version that we only need to replace everything that contains "StartTime" to be "". but, if we want to re-use the queryValidator, there'll be more parameters that needed to be passed in, and there'll be more branches, which can make the code more complicated. 

We can make the queryValidator to be an interface so that this filter can implement the interface, but given that this is only a temporary component, I think that this might be an overkill?

Reply:
Thanks for the explanation! Yeah, it might be too much to change the existing validator. We need to deprecate this after we remove ES anyway. You can add what you said above in the comment and add one more line says "Will be removed after fully operate on Pinot".